examples/vnet/doca_gw_dpdk.c

```diff
@@ -0,0 +1,338 @@
+	}
+
+	if (set_item_value) {
+		item->spec = spec;
+		item->mask = mask;
+		if (entry->flags)
+			entry->apply = doca_gw_dpdk_modify_ipv6_item;
+	}
+}
+
+static int doca_gw_dpdk_modify_vxlan_item(__rte_unused struct doca_dpdk_item_entry *entry,
+						__rte_unused struct doca_gw_match *match)
+{
+	return 0;
+}
+
+static void doca_gw_dpdk_build_vxlan_flow_item(struct doca_dpdk_item_entry *entry,
+			struct doca_gw_match *match)
+{
+	struct rte_flow_item *flow_item = entry->item;
+	struct rte_flow_item_vxlan *spec = &entry->item_data.vxlan.spec;
+	struct rte_flow_item_vxlan *mask = &entry->item_data.vxlan.mask;
+
+	flow_item->type = RTE_FLOW_ITEM_TYPE_VXLAN;
+	if (!match->tun.vxlan.tun_id) 
+		return;
+	memcpy(spec->vni, (uint8_t*)(&match->tun.vxlan.tun_id), 3); //TODO
+	flow_item->spec = spec;
+	flow_item->mask = mask;
+	if (match->tun.vxlan.tun_id == UINT32_MAX) {
+		entry->flags |= DOCA_SET_VLAN_ID;
+		entry->apply = doca_gw_dpdk_modify_vxlan_item;
+	}
+}
+
+static void doca_gw_dpdk_build_tcp_flow_item(struct doca_dpdk_item_entry *entry,
+			struct doca_gw_match *match, uint8_t type)
+{
+	struct rte_flow_item *item = entry->item;
+	struct rte_flow_item_tcp *spec = &entry->item_data.tcp.spec;
+	struct rte_flow_item_tcp *mask = &entry->item_data.tcp.mask;
+
+	item->type = RTE_FLOW_ITEM_TYPE_TCP;
+	spec->hdr.src_port = (type == INNER_MATCH)? match->in_src_port : match->out_src_port;
+	spec->hdr.dst_port = (type == INNER_MATCH)? match->in_dst_port : match->out_dst_port;
+	item->spec = spec;
+	item->mask = mask;
+	//no need modify those fields.
+	return ;
+}
+
+static void doca_gw_dpdk_build_udp_flow_item(struct doca_dpdk_item_entry *entry,
+			struct doca_gw_match *match, uint8_t type)
+{
+	struct rte_flow_item *item = entry->item;
+	struct rte_flow_item_udp *spec = &entry->item_data.udp.spec;
+	struct rte_flow_item_udp *mask = &entry->item_data.udp.mask;
+
+	item->type = RTE_FLOW_ITEM_TYPE_UDP;
+	spec->hdr.src_port = (type == INNER_MATCH)? match->in_src_port : match->out_src_port;
+	spec->hdr.dst_port = (type == INNER_MATCH)? match->in_dst_port : match->out_dst_port;
+	item->spec = spec;
+	item->mask = mask;
+	//no need modify those fields.
+	return ;
+}
+
+static int doca_gw_dpdk_build_item(struct doca_gw_match *match,
+					struct doca_gw_pipe_dpdk_flow *pipe_flow, struct doca_gw_error *err)
+{
+#define NEXT_ITEM (&pipe_flow->item_entry[idx++]) //reduce line length
+	uint8_t idx = 0, type = OUTER_MATCH;
+
+	doca_gw_dpdk_build_eth_flow_item(NEXT_ITEM, match);
+	if (match->vlan_id) //0: not set; U16_MAX: fixed or need modify ; others: fixed
+		doca_gw_dpdk_build_vlan_item(NEXT_ITEM, match);
+	if (doca_match_is_ipv4(match, type))
+		doca_gw_dpdk_build_ipv4_flow_item(NEXT_ITEM, match, type);
+	else
+		doca_gw_dpdk_build_ipv6_flow_item(NEXT_ITEM, match, type);
+	if((match->tun.type != DOCA_TUN_NONE)) {
+		switch (match->tun.type) {
+		case DOCA_TUN_VXLAN:
+			doca_gw_dpdk_build_udp_flow_item(NEXT_ITEM, match, type);
```

Comment:
this is ok, but maybe we want to fill it our self it user didn't out udp and destination port